prototype2/common/StatPublisher.h

```diff
@@ -11,16 +11,33 @@
 
 #include <common/Detector.h>
 #include <libs/include/Socket.h>
+#include <libs/include/Timer.h>
 #include <string>
 
 class StatPublisher {
 public:
-  /// Connect to a Carbon/Graphite server bu ip address and tcp port
+  /// \brief Connect to a Carbon/Graphite server bu ip address and tcp port
   StatPublisher(std::string ip, int port);
 
-  /// Send detector metrics to Carbon/Graphite server
+  /// \brief Send detector metrics to Carbon/Graphite server
   void publish(std::shared_ptr<Detector> detector);
 
+  /// \brief called when senddata() fails
+  void handleReconnect();
+
+  /// \brief
+  void reconnectHelper();
+
 private:
+  /// Connection variable
```

Comment:
`statdb` should really be of type `std::unique_ptr<TCPTransmitter>` to prevent manual memory management and leaks. There is one (a leak) now. Furthermore, with `statdb` should be renamed (to `StatDB` maybe?) in order to follow the LLVM coding standard.